modules/repofiles/update.go

```diff
@@ -495,9 +495,18 @@ func PushUpdate(repo *models.Repository, branch string, opts PushUpdateOptions)
 		return err
 	}
 
-	log.Trace("TriggerTask '%s/%s' by %s", repo.Name, branch, pusher.Name)
+	if opts.NewCommitID != git.EmptySHA {
+		if err = models.RemoveDeletedBranch(repo.ID, opts.Branch); err != nil {
```

Comment:
But if the new commit id is not empty the branch has not been deleted...

Why are you deleting it here?

Reply:
see https://github.com/go-gitea/gitea/pull/9927#discussion_r369635383